esphome/components/uart/switch/__init__.py

```diff
@@ -9,17 +9,34 @@
 
 UARTSwitch = uart_ns.class_("UARTSwitch", switch.Switch, uart.UARTDevice, cg.Component)
 
+CONF_DATA_ON = "data_on"
+CONF_DATA_OFF = "data_off"
 
-CONFIG_SCHEMA = (
+
+def validate_data(config):
+    if (CONF_DATA_ON in config or CONF_DATA_OFF in config) and (
+        CONF_DATA in config or CONF_SEND_EVERY in config
+    ):
+        raise cv.Invalid(
+            '"data" or "send_every" can\'t be used with "data_on" or "data_off"'
+        )
+    return config
+
+
+CONFIG_SCHEMA = cv.All(
     switch.switch_schema(UARTSwitch, block_inverted=True)
     .extend(
         {
-            cv.Required(CONF_DATA): validate_raw_data,
+            cv.Optional(CONF_DATA): validate_raw_data,
+            cv.Optional(CONF_DATA_OFF): validate_raw_data,
+            cv.Optional(CONF_DATA_ON): validate_raw_data,
```

Comment:
If desired, something like this could be better.
```yaml
switch:
  - platform: uart
    data: "blah"
  - platform: uart
    data:
      on: "on"
      off: "off"
```

```suggestion
            cv.Optional(CONF_DATA): cv.Any(
                validate_raw_data, 
                cv.Schema({
                    cv.Optional(CONF_OFF): validate_raw_data,
                    cv.Optional(CONF_ON): validate_raw_data,
                }),
            )
```

Reply:
oh...  that's nice!